Add blog post about storage resizing (#49)
strimzi/strimzi.github.io: _posts/2019-03-01-resizing-persistent-volumes.md

tmpfs           3.9G     0  3.9G   0% /proc/acpi
tmpfs           3.9G     0  3.9G   0% /proc/scsi
tmpfs           3.9G     0  3.9G   0% /sys/firmware
```

Notice the volume `/dev/xvdbe` which is the persistent volume with the Kafka message logs and has total capacity of 99G.
As you can see I have still a lot of free disk space.
But I want to increase the disk size anyway.
Let's say to 200Gi.

As the first step, I have to go to the Persistent Volume Claims and edit the requested capacity.
In my case I will change the requested capacity from 100Gi to 200Gi.
You can apply this change to all PVC of your Kafka cluster.

```yaml
# ...
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 200Gi
  volumeName: pvc-68b4be67-3a8a-11e9-bf2b-067919d3dbb8
  storageClassName: gp2
  dataSource: null
# ...
```

After you apply these changes, the `status` field of the PVC should change to something like this:

```yaml
#...
status:
  phase: Bound
  accessModes:
    - ReadWriteOnce
  capacity:
    storage: 100Gi
  conditions:
    - type: Resizing
      status: 'True'
      lastProbeTime: null
      lastTransitionTime: '2019-02-27T12:25:15Z'
```

> As you can see, the capacity is still 100Gi, but Kubernetes / OpenShift are now resizing the volume.

Reviewer: ```suggestion
As you can see, the capacity is still 100Gi, but Kubernetes / OpenShift is now resizing the volume.
```

Once the resizing of the volume is finished, the `status` will change to this:

```yaml
# ...
status:
  phase: Bound
  accessModes:
    - ReadWriteOnce
  capacity:
    storage: 100Gi
  conditions:
    - type: FileSystemResizePending
      status: 'True'
      lastProbeTime: null
      lastTransitionTime: '2019-02-27T12:25:30Z'
      message: >-
        Waiting for user to (re-)start a pod to finish file system resize of
        volume on node.
```

As you can read in the message, the volume has been resized.
But the file system is still only 100Gi big and needs to be resized as well.
The resizing of the file system cannot be done while the pod is running - it will be done automatically when the pod is restarted. 

> The easiest way how to restart the Kafka pods to resize the file system is to tell Strimzi to perform a rolling update of the Kafka Statefulset.

Reviewer: ```suggestion
The easiest way to restart the Kafka pods to resize the file system is to tell Strimzi to perform a rolling update of the Kafka Statefulset.
```

You can do that using the [`strimzi.io/manual-rolling-update=true`annotation](https://strimzi.io/docs/latest/full.html#proc-manual-rolling-update-kafka-deployment-configuration-kafka).

```console
kubectl annotate statefulset cluster-name-kafka strimzi.io/manual-rolling-update=true
```

> After you set this annotation, Strimzi will roll all the Kafka pods one by one.

Reviewer: Mention that it might take upto a reconciliation interval for this to happen because the CO is not watching the SS?

While the pods are being restarted, the filesystem will be resized.

> In case you already run out of disk space and your Kafka pods are in the crash loop, you might need to delete the pods manually.

Reviewer: ```suggestion
In case you already ran out of disk space and your Kafka pods are crash-looping, you might need to delete the pods manually.
```

If you check your PVCs after the restart is finished, you should see something like this:

```yaml
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  annotations:
    pv.kubernetes.io/bind-completed: 'yes'
    pv.kubernetes.io/bound-by-controller: 'yes'
    strimzi.io/delete-claim: 'false'
    volume.beta.kubernetes.io/storage-provisioner: kubernetes.io/aws-ebs
  selfLink: /api/v1/namespaces/my-kafka/persistentvolumeclaims/data-my-cluster-kafka-0
  resourceVersion: '45337'
  name: data-my-cluster-kafka-0
  uid: 68b4be67-3a8a-11e9-bf2b-067919d3dbb8
  creationTimestamp: '2019-02-27T12:22:54Z'
  namespace: my-kafka
  finalizers:
    - kubernetes.io/pvc-protection
  labels:
    strimzi.io/cluster: my-cluster
    strimzi.io/kind: Kafka
    strimzi.io/name: my-cluster-kafka
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 200Gi
  volumeName: pvc-68b4be67-3a8a-11e9-bf2b-067919d3dbb8
  storageClassName: gp2
  dataSource: null
status:
  phase: Bound
  accessModes:
    - ReadWriteOnce
  capacity:
    storage: 200Gi
```

> As you can see, the requested size is no 200Gi and the actual capacity is now also 200Gi.

Reviewer: ```suggestion
As you can see, the requested size is now 200Gi and the actual capacity is also now 200Gi.
```
